src/uthash.h

```diff
@@ -26,9 +26,11 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 #define UTHASH_VERSION 2.0.1
 
+#ifndef UTHASH_FREESTANDING_IMPLEMENTATION
 #include <string.h>   /* memcmp,strlen */
 #include <stddef.h>   /* ptrdiff_t */
 #include <stdlib.h>   /* exit() */
```

Comment:
My next rabbit-hole question here is: if you skip including these headers, then where do you get the `NULL` macro from? Is there a more freestanding-friendly way for us to get it?

FYI I'm still looking into the `uint8_t`, `uint32_t`, `ptrdiff_t` question, but my gut instinct is that I *don't* want to adopt your proposed `uthash_uint8_t` stuff; it seems really un-simple.